pkg/ipamd/ipamd.go

```diff
@@ -562,13 +591,14 @@ func (c *IPAMContext) tryFreeENI() {
 	}
 }
 
-// tryUnassignIPsFromAll determines if there are IPs to free when we have extra IPs beyond the target and warmIPTargetDefined
+// tryUnassignIPsorPrefixesFromAll determines if there are IPs to free when we have extra IPs beyond the target and warmIPTargetDefined
 // is enabled, deallocate extra IP addresses
-func (c *IPAMContext) tryUnassignIPsFromAll() {
-	if _, over, warmIPTargetDefined := c.ipTargetState(); warmIPTargetDefined && over > 0 {
+func (c *IPAMContext) tryUnassignIPsOrPrefixesFromAll() {
```

Comment:
Consider keeping them separate. These are two different (similar looking) workflows but something you can do without touching the existing functions. Instead of `tryUnassignIPsOrPrefixesFromAll`, you can have `tryUnassignIPsFromAll` and `tryUnassignPrefixesFromAll` and you won't have to pass the `c.enableIpv4PrefixDelegation` flag all the way through functions and you won't have overload variables based on the `c.enableIpv4PrefixDelegation` flag. It'll be simpler and easier to maintain. You could, of course merge, any common functionality into a helper if there's a need

Reply:
Yes, I had it as 2 separate functions but based on internal feedback I merged them. Since even I wanted to keep the existing functions untouched. Sure will separate it out.

Reply:
Discussed with Fawad offline, will see how I can restructure this. Mainly if we have 2 functions `tryUnassignIPsFromAll` and  `tryUnassignPrefixesFromAll`, both functions will be doing very similar tasks of getting the current warm setting, get free IPs/prefixes, clean up datastore and dealloc the prefix/IP so instead I pushed the logic of returning a IP or prefix to the final API - `findFreeableIPsOrPrefixes` which does - 

```
if !c.enableIpv4PrefixDelegation {
		freeableIPs = c.dataStore.FreeableIPs(eni)
	} else if c.enableIpv4PrefixDelegation {
		freeableIPs = c.dataStore.FreeablePrefixes(eni)
	}
```